lib/Test/MockFile.pm

```diff
@@ -1712,6 +1714,120 @@ BEGIN {
         $mock->{'has_content'} = undef;
         return 1;
     };
+
+    *CORE::GLOBAL::chown = sub(@) {
+        my ( $uid, $gid, @files ) = @_;
+
+        $^O eq 'MSWin32'
+            and return 0; # does nothing on Windows
+
+        # Not an error, report we changed zero files
+        @files
+            or return 0;
+
+        my %mocked_files = map +( $_ => _get_file_object($_) ), @files;
+
+        # The idea is that if some are mocked and some are not,
+        # it's probably a mistake
+        my $mocked_count = grep ref, values %mocked_files;
+        if ( $mocked_count && $mocked_count != @files ) {
+            die "You called chown() on a mix of mocked and unmocked files - this is very likely a bug on your side";
+        }
+
+        my $num_changed = 0;
+        foreach my $file (@files){
+            my $mock = $mocked_files{$file};
+
+            if ( !$mock ) {
+                _real_file_access_hook( 'chown', \@_ );
+                goto \&CORE::chown if _goto_is_available();
+                return CORE::chown(@files);
+            }
+
+            if ( !$mock->exists() ) {
+                if ( $mock->is_file() ) {
+                    $! = ENOENT;
+                }
+
+                if ( $mock->is_dir() ) {
+                    $! = ENOTDIR;
```

Comment:
If the path doesn't exist, it's always going to be the same error from perl's perspective, right?

```
$>perl -E'say chown (0, 0, qw{/nofile /notafile }); say $!+0; say "$!"'
0
2
No such file or directory
```

Reply:
2 seems to be `ENOENT`?

Reply:
I was certain I tested it with `foo/` (ending with a forward slash) and it gave a different error but I can't replicate it, so I guess I saw it elsewhere and mistook it for this scenario. I'll correct this. Thanks.